### PR TITLE
* DDO-1259 Make charts linter compliant

### PR DESCRIPTION
**Pull request by knapii-developments**
We're adding linting policies that deployment manifest must adhere to. In order for the action to succeed the charts need to be updated. 

Here's the succeeding action [here](https://github.com/broadinstitute/terra-helmfile/pull/1324/checks?check_run_id=2886060189#step:7:8)

### CODE REVIEW

**Review by choover-broad — COMMENTED** [review]
Nice work @knapii-developments! The only potentially breaking changes I see are:
* rename of the Cromwell deployment
* rename of the SAs for WSM and Buffer

I _think_ the SA renames will have no impact.

But the deployment rename will cause the existing Cromwell deployments to be deleted and recreated, leading to greater disruption during the release than usual. It might make sense for us to ship these changes outside the regular release cycle, during a maintenance window.

File: charts/workspacemanager/templates/deployment.yaml [hunk 1]
```diff
@@ -36,7 +36,7 @@ spec:
         checksum/{{ .Values.name }}-prometheus-configmap: {{ include (print $.Template.BasePath "/configmaps/prometheus-configmap.yaml") . | sha256sum }}
         {{- end }}
     spec:
-      serviceAccountName: {{ .Values.name }}-service-sa
+      serviceAccountName: {{ .Values.name }}-sa
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/workspacemanager/templates/deployment.yaml:39]: The only place I see this for workspace manager is in a [service account manifest](https://github.com/broadinstitute/terra-helm/blob/master/charts/workspacemanager/templates/service-account.yaml). Should I be linting against those?

File: charts/buffer/templates/deployment.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ spec:
         checksum/{{ .Values.name }}-proxy-configmap: {{ include (print $.Template.BasePath "/configmaps/proxy-apache-httpd-configmap.yaml") . | sha256sum }}
         checksum/{{ .Values.name }}-tcell-configmap: {{ include (print $.Template.BasePath "/configmaps/proxy-tcell-configmap.yaml") . | sha256sum }}
     spec:
-      serviceAccountName: {{ .Values.name }}-service-sa
+      serviceAccountName: {{ .Values.name }}-sa
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/buffer/templates/deployment.yaml:33]: I didn't realize that we should be changing these for service accounts. However,  made the changes here f6f58cf

File: charts/cromwell/templates/_deployment.tpl [hunk 3]
```diff
@@ -29,8 +29,11 @@ spec:
       annotations:
         {{- /* Automatically restart deployments on config map change: */}}
         checksum/{{ $settings.name }}-cm: {{ $outputs.configmapChecksum }}
+        {{- range $key, $value := $settings.annotations }}
+        {{ $key }}: {{ $value }}
+        {{- end }}
     spec:
-      serviceAccountName: cromwell-sa
+      serviceAccountName: {{ $settings.name }}-sa
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/cromwell/templates/_deployment.tpl]: f6f58cf

File: charts/workspacemanager/templates/deployment.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/workspacemanager/templates/deployment.yaml:39]: 71d3e89 and 9fe1c4d

File: charts/leonardo/templates/deployments/_deployment.tpl [hunk 4]
```diff
@@ -29,8 +29,11 @@ spec:
       annotations:
         {{- /* Automatically restart deployments on config map change: */}}
         checksum/{{ $settings.name }}-cm: {{ $outputs.configmapChecksum }}
+        {{- range $key, $value := $settings.annotations }}
+        {{ $key }}: {{ $value }}
+        {{- end }}
     spec:
-      serviceAccountName: leonardo-sa
+      serviceAccountName: {{ $settings.name }}-sa
```
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/leonardo/templates/deployments/_deployment.tpl]: f6f58cf

File: charts/rawls/templates/_deployment.tpl [hunk 5]
```diff
@@ -29,8 +29,11 @@ spec:
       annotations:
         {{- /* Automatically restart deployments on config map change: */}}
         checksum/{{ $settings.name }}-cm: {{ $outputs.configmapChecksum }}
+        {{- range $key, $value := $settings.annotations }}
+        {{ $key }}: {{ $value }}
+        {{- end }}
     spec:
-      serviceAccountName: rawls-sa
+      serviceAccountName: {{ $settings.name }}-sa
```
Comment (choover-broad) [inline charts/rawls/templates/_deployment.tpl:36]: `rawls-sa`

File: charts/ncbiaccess/templates/deployment.yaml [hunk 6]
```diff
@@ -17,6 +17,10 @@ spec:
   template:
     metadata:
       labels: {{- include "ncbiaccess.labels" . | nindent 8 }}
+      annotations:
+          {{- range $key, $value := .Values.annotations }}
```
[the remark this answers is not in the document]
Comment (rmeffan) [reply charts/ncbiaccess/templates/deployment.yaml:21]: Clearly, we need a linter to lint linter changes.

**Review by rmeffan — APPROVED** [review]
LGTM overall. I don't know enough about the underlying process to comment on the technical merits, but Chelsea is also reviewing it, so you're in good hands.

File: charts/cromwell/templates/_deployment.tpl [hunk 7]
```diff
@@ -7,7 +7,7 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ $settings.name }}
+  name: {{ $settings.name }}-deployment
```
Comment (choover-broad) [inline charts/cromwell/templates/_deployment.tpl:10]: We were going to get rid of this rename and an exception for Cromwell, right @knapii-developments?

**Review by choover-broad — APPROVED** [review]
Approved, left a few comments in-line that need to be addressed but otherwise I think this is set!

File: charts/cromwell/templates/_deployment.tpl (hunk 7 above)
Comment (knapii-developments) [reply charts/cromwell/templates/_deployment.tpl]: Oh I thought it was for the service accounts. 04b7e36

File: charts/ncbiaccess/templates/deployment.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (knapii-developments) [reply charts/ncbiaccess/templates/deployment.yaml:21]: 3c5469a

**Pull request merged**